test/fixtures/simulated_ci_environment/main.tf

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+provider "google" {
+  project = "${var.project_id}"
+  region  = "${var.region}"
+}
+
+locals {
+  required_service_account_project_roles = [
+    "roles/compute.instanceAdmin",
+    "roles/iam.serviceAccountUser",
+  ]
+}
+
+resource "google_project" "startup_scripts" {
+  name            = "startup-scripts"
+  project_id      = "startup-scripts"
+  folder_id       = "${var.folder_id}"
+  billing_account = "${var.billing_account}"
+}
+
+resource "google_project_services" "startup_scripts" {
```

Comment:
+0: Something to consider here is using `google_project_service` with a count - that will manage services in an additive manner rather than an authoritative manner. This'll help avoid issues like the flapping "oslogin" API.

Reply:
Great idea, I refactored this in 5cdbb87b47ff1de55c6f277c811ed7b7e969a8e8

On a related note, I see there seems to be implicit dependencies used as a form of explicit dependencies, specifically how `project = "${google_project_services.startup_scripts.project}"` was used in the `google_project_iam_binding` resource.

I left them mostly as is for now, but something I'll be keeping an eye out for in the future.  I suggest we try and move implicit dependencies which seem to be expressed explicitly, e.g. using the `project` attribute to explicit ones using `depends_on`.

This does however run into https://github.com/hashicorp/terraform/issues/15285 where you can't explicitly depend on a count instance, but you can depend on the whole collection, which may have 0 instances.